src/Internal/Test/E2E/Runner.purs

```diff
@@ -265,12 +273,20 @@ testPlan opts@{ tests } rt@{ wallets } =
                 , confirmAccess: confirmAccess extensionId re
                 , sign: sign extensionId password re
                 }
-            subscribeToBrowserEvents (Just $ Seconds 10.0) page
-              case _ of
-                ConfirmAccess -> launchAff_ someWallet.confirmAccess
-                Sign -> launchAff_ someWallet.sign
-                Success -> pure unit
-                Failure err -> throw err
+            makeAff $ \k -> do
+              let
+                rethrow aff = launchAff_ do
+                  res <- try aff
+                  when (isLeft res) $ liftEffect $ k res
+
+              map fiberCanceler $ launchAff $ (try >=> k >>> liftEffect) $
+                subscribeToBrowserEvents (Just $ Seconds 10.0) page
+                  case _ of
+                    ConfirmAccess -> rethrow someWallet.confirmAccess
+                    Sign -> rethrow someWallet.sign
+                    Success -> pure unit
```

Comment:
Shouldn't we pass the `unit` to `k` callback to make it exit successfully? Have you tried to run manually the e2e test suite for all these cases (error from sign, error from confirm, success, failure)?

Reply:
I think not. Because we should pass result of the whole computation, to continue afterwards (so result of `subscribeToBrowserEvents`, and we do pass it to k). The "case" snippet gets called multiple times inside `subscribeToBrowserEvents`. Changing it to:
```
case _ of 
  Success -> k unit
``` 
would shortcircuit the computation and exit once this handler is called. Simmilar with `Failure -> throw Error` (but see below comment).

### testing
I wasn't able to run e2e tests succesfully, so i havent verified that succesful tests remain succesful.

So i tested like this: 
1. i substituted `throwError` for `confirmAccess`
2. `confirmAccess` then errored before the usual errors with my supplied error
3. it errors the test, not the whole test suite (in contrary to previous commit which i tested as well)

We could wait to first fix e2e tests and then Ill test this change again, but i think this works as intended